packages/shared/src/components/auth/RegistrationForm.tsx

```diff
@@ -242,6 +243,18 @@ const RegistrationForm = ({
     !isSubmitted || !hints?.['traits.experienceLevel'];
   const { isAuthenticating = false } = useAtomValue(authAtom);
 
+  const usernameIcon = (() => {
+    if (isLoading) {
+      return <Loader />;
```

Comment:
Would be cool if it was some custom loader like "generating your amazing username"
Feel we decided to not block this so users could just type whatever they want?